build/Makefile

```diff
@@ -0,0 +1,22 @@
+lisk_version := $(shell jq --raw-output .version ../package.json)
+
+release/lisk-$(lisk_version)-Linux-x86_64.tar.gz: check src/lisk-$(lisk_version).tgz
+	bash ./build.sh -n $(LISK_NET) -v $(lisk_version) -c
+
+.PHONY: check
+check:
+ifndef LISK_NET
+	$(error The LISK_NET environment variable must be set.)
+endif
+
+src/lisk-$(lisk_version).tgz: ../REVISION
+	cd ..; tar czf build/src/lisk-$(lisk_version).tgz \
+	    --exclude-vcs --exclude=build --exclude-ignore=build/build.ignore \
+	    --transform='s#^./#package/#' .
```

Comment:
It seems you completely removed the behavior of `npm pack` with the file `build.ignore`. I feel it's not good practice. You should use `.npmignore` file as mostly developers will not aware of `build.ignore` file. For developer point of view they will always modify/look `.npmignore` to see which files will be shipped in release.

Reply:
While we do not use the same tool (`tar` instead of `npm pack`) or ignore list the behaviour is the same: we create the same file in the same format . It is true that we ignore a few more files though.
The reason we do not use the existing `.npmignore` is because its format is slightly different (and not documented?) -- do we need it at all if we never call `npm pack`?

Reply:
@fchavant That's not true exactly, Just check this out https://docs.npmjs.com/misc/developers

> .npmignore files follow the same pattern rules as .gitignore files:
> 
> Blank lines or lines starting with # are ignored.
> Standard glob patterns work.
> You can end patterns with a forward slash / to specify a directory.
> You can negate a pattern by starting it with an exclamation point !.
> .
> .
> Additionally, everything in node_modules is ignored, except for bundled dependencies. npm automatically handles this for you, so don’t bother adding node_modules to .npmignore.
> 

So `.npmignore` is very well documented and actually follow the exact rules of `.gitingore`. 

Now just consider a scenario that you ask any Core developer (any NodeJS developer):

- Which files will be shipped in production? 
- A developer created a file, but he don't want it to be shipped in production. 

Everyone will go and look for `.npmignore` file. So removing a file which have so much significance in technology domain is not right approach. 

The problem during the build is that you don't want to use `npm pack`, so you come up with a workaround for it, but by not removing `.npmignore` file.

Reply:
So what is your suggestion?

Reply:
Two possibilities. 

1. Use `npm pack` to we can use its power and also benefit if it introduce new feature in future. 
2 Use custom bash script to strip down the files using `.npmignore` and then pass it to tar.

Reply:
Now it's often mainly use 
https://docs.npmjs.com/files/package.json#files